src/utils.ts

```diff
@@ -3357,58 +3979,115 @@ export class ParallelAxis extends newAxis {
 }
 
 export class ShapeCollection {
+  public minimum: Vertex;
+  public maximum: Vertex;
   constructor(
-    public drawings: newShape[] = [],
-    public frame: DOMMatrix = new DOMMatrix()
-  ) {}
+    public shapes: newShape[] = [],
+  ) {
+    [this.minimum, this.maximum] = this.getBounds();
+  }
+
+  public get length(): number { return this.shapes.length }
+
+  public includes(shape: newShape) { return this.shapes.includes(shape) }
+
+  public static fromPrimitives(primitives: { [key: string]: any }, scale: Vertex = new Vertex(1, 1)): ShapeCollection {
+    const shapes = [];
+    primitives.forEach(primitive => shapes.push(newShape.deserialize(primitive, scale)))
```

Comment:
shapes could be computed with a map instead of a forEach + loop

Reply:
Thanks